### ISSUE TITLE
[Music.py] Loop doesn't work after commit 6467061

### ISSUE THREAD

**Issue by maj113** [maj113/Angel6#51]
guilty commit removed the `if not self.loop:` which is probably what broke it, have to experiment further.
It could also have been me using `asyncio.gather` since it doesn't block the function

**Comment by maj113** [comment]
reverting back to this fixed the issue


```py

    async def audio_player_task(self):
        """Plays audio from the queued songs continuously in the background until stopped.

        Dequeues the next song from the queue, creates a playable audio stream from the song's source
        URL using FFmpeg, and starts playing the audio through the voice connection. If looping is
        disabled, sends an embed message to the channel indicating that the current song is playing.
        """
        while True:
            self.next.clear()  # FIXME: significant performance slowdowns here
            self.now = None
            if not self.loop:
                # Try to get the next song within 3 minutes.
                # If no song will be added to the queue in time,
                # the player will disconnect due to performance
                # reasons.
                try:
                    async with asyncio.timeout(180):  # 3 minutes
                        self.current = (
                            await self.songs.get()
                        )  # NOTICE: this must not be called when looping is enabled
                except asyncio.TimeoutError:
                    self.bot.loop.create_task(self.stop())
                    self.exists = False
                    return

                self.now = await discord.FFmpegOpusAudio.from_probe(
                    self.current.source.stream_url, **YTDLSource.FFMPEG_OPTIONS
                )
                self.voice.play(self.now, after=self.play_next_song)
                await self.current.source.channel.send(
                    embed=self.current.create_embed()
                )
            # If the song is looped
            elif self.loop:
                self.now = await discord.FFmpegOpusAudio.from_probe(
                    self.current.source.stream_url, **YTDLSource.FFMPEG_OPTIONS
                )
                self.voice.play(self.now, after=self.play_next_song)

            await self.next.wait()
            
```

**Comment by maj113** [comment]
however now im left with code duplication and the inability to use asyncio.gather here, ill have to investigate further

**Comment by maj113** [comment]
fixed in 2.4.0
